File: CausalInference/Udemy_Econometrics_and_Statistics_for_Business_in_R_n_Python/2. Google_s Causal Impact/Causal Impact.R
#define pre and post period dates
start = "2016-01-01"
treatment = "2018-03-17"
end = "2018-07-17"

#retrieve data
#install.packages("tseries")
library(tseries)
Facebook <- get.hist.quote(instrument = "FB",
                           start = start,
                           end = end,
                           quote = "Close",
                           compression = "w")
Walmart <- get.hist.quote(instrument = "WMT",
                           start = start,
                           end = end,
                           quote = "Close",
                           compression = "w")
Disney <- get.hist.quote(instrument = "DIS",
                           start = start,
                           end = end,
                           quote = "Close",
                           compression = "w")
BMW <- get.hist.quote(instrument = "BMW.DE",
                           start = start,
                           end = end,
                           quote = "Close",
                           compression = "w")
Novartis <- get.hist.quote(instrument = "NVS",
                           start = start,
                           end = end,
                           quote = "Close",
                           compression = "w")
Goldman_sachs <- get.hist.quote(instrument = "GS",
                           start = start,
                           end = end,
                           quote = "Close",
                           compression = "w")
GE <- get.hist.quote(instrument = "GE",
                           start = start,
                           end = end,
                           quote = "Close",
                           compression = "w")
Heinz <- get.hist.quote(instrument = "KHC",
                           start = start,
                           end = end,
                           quote = "Close",
                           compression = "w")
McDonalds <- get.hist.quote(instrument = "MCD",
                           start = start,
                           end = end,
                           quote = "Close",
                           compression = "w")
Carlsberg <- get.hist.quote(instrument = "CARL-B.CO",
                           start = start,
                           end = end,
                           quote = "Close",
                           compression = "w")


#plotting data
series <- cbind(Facebook, Walmart, Disney, BMW, Novartis, Goldman_sachs,
                GE, Heinz, McDonalds, Carlsberg)
series <- na.omit(series)
#install.packages("ggplot2")
library(ggplot2)
autoplot(series, facet = NULL) + xlab("time") + ylab("Price Close")

#correlation check
dataset_cor <- window(series, start = start, end = treatment)
dataset_cor <- as.data.frame(dataset_cor)
cor(dataset_cor)

#selecting the final dataset
final_series <- cbind(Facebook, Walmart, Goldman_sachs, McDonalds, Carlsberg)
final_series <- na.omit(final_series)

#Create pre and post period objects
pre.period <- as.Date(c(start, treatment))
post.period <- as.Date(c(treatment, end))

#Running Causal Impact
#install.packages("CausalImpact")
library(CausalImpact)
impact <- CausalImpact(data = final_series,
                       pre.period = pre.period,
                       post.period = post.period,
                       model.args = list(niter = 2000,
                                         nseasons = 52))

#Visualize results
plot(impact)
summary(impact)
summary(impact, "report")

File: CausalInference/Udemy_Econometrics_and_Statistics_for_Business_in_R_n_Python/2. Google_s Causal Impact/Causal Impact.py
#import libraries
import yfinance as yf

#define dates
start = "2015-12-28"
training_end = "2018-03-05"
treatment_start = "2018-03-12"
end = "2018-07-16"
end_stock = "2018-07-17"

#get stock data
stocks = ["FB", "WMT", "DIS", "BMW.DE", "NVS",
          "GS", "GE", "KHC", "MCD", "CARL-B.CO"]
dataset = yf.download(stocks,
                      start = start,
                      end = end_stock,
                      interval = '1wk')

#getting only the data we want
dataset = dataset.iloc[:, :10]

#getting rid of unneeded level
dataset.columns = dataset.columns.droplevel()

#getting rid of NAs
dataset = dataset.dropna()

#Plotting data
dataset.plot()

#Create dataset with training period only
dataset_cor = dataset[dataset.index <= treatment_start]

#correlation matrix
dataset_cor.corr()
import pandas as pd
pd.set_option('display.max_columns', None)

#create dataset with the most correlated
final_series = dataset[["FB", "CARL-B.CO", "GS", "MCD", "NVS", "WMT"]]

#create pre and post period
pre_period = [start, training_end]
post_period = [treatment_start, end]

#import library
from causalimpact import CausalImpact
impact = CausalImpact(data = final_series,
                      pre_period = pre_period,
                      post_period = post_period,
                      nseasons = [{'period': 52}],
                      priod_level_sd = None)

#interpret the results
impact.plot()
print(impact.summary())
print(impact.summary('report'))












































































